registry/storage/driver/gdrive/gdrive.go

```diff
@@ -0,0 +1,497 @@
+// Package gdrive provides a storagedriver.StorageDriver implementation to
+// store blobs in Google drive storage.
+//
+// This package leverages the google.golang.org/api/drive/v3 client library
+// for interfacing with google drive.
+//
+// Parameters :
+//
+// keyFile : A private service account key file in JSON file format that can be downloaded
+//           from google api console.
+// rootdirectory : Folder name in google drive to store all registry files.
+//
+// +build include_gdrive
+
+package gdrive
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/base"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/jwt"
+	drive "google.golang.org/api/drive/v3"
+)
+
+const driverName = "gdrive"
+
+// DriverParameters represents all configuration options available for the
+// filesystem driver
+type DriverParameters struct {
+	KeyFile       string
+	RootDirectory string
+}
+
+func init() {
+	factory.Register(driverName, &gdriveDriverFactory{})
+}
+
+// gdriveDriverFactory implements the factory.StorageDriverFactory interface
+type gdriveDriverFactory struct{}
+
+func (factory *gdriveDriverFactory) Create(parameters map[string]interface{}) (storagedriver.StorageDriver, error) {
+	return FromParameters(parameters)
+}
+
+type driver struct {
+	rootID       string
+	driveService *drive.Service
+}
+
+type baseEmbed struct {
+	base.Base
+}
+
+// Driver is a storagedriver.StorageDriver implementation backed by Google Drive.
+type Driver struct {
+	baseEmbed
+}
+
+// FromParameters constructs a new Driver with a given parameters map
+func FromParameters(parameters map[string]interface{}) (*Driver, error) {
+	params, err := fromParametersImpl(parameters)
+	if err != nil || params == nil {
+		return nil, err
+	}
+	return New(*params)
+}
+
+func fromParametersImpl(parameters map[string]interface{}) (*DriverParameters, error) {
+
+	keyFile, ok := parameters["keyFile"]
```

Comment:
Don't use camel case: we only use flat in configuration for registry: `keyfile`.